vite.config.js

```diff
@@ -4,11 +4,24 @@ import react from '@vitejs/plugin-react';
 import {vanillaExtractPlugin} from '@vanilla-extract/vite-plugin';
 import noBundlePlugin from 'vite-plugin-no-bundle';
 import preserveDirectivesPlugin from 'rollup-plugin-preserve-directives';
+import {version as packageVersion} from './package.json';
+
+/**
+ * Identifiers must start with a letter (not number) and can't contain dots
+ * Removing trailing zeros to reduce the size of the identifier
+ *
+ * 1.2.3 => v1_2_3
+ * 1.2.0 => v1_2
+ * 1.0.0 => v1
+ */
+const version = 'v' + packageVersion.replace(/(\.0)*$/, '').replace(/\./g, '_');
 
 export default defineConfig({
     plugins: [
         react(),
-        vanillaExtractPlugin(),
+        vanillaExtractPlugin({
+            identifiers: ({hash}) => `${version}_${hash}`,
```

Comment:
another idea:

Create a map hash=>autoincrementCounter

and the resulting identifiers would be: `v1_2_3_i455` to make it even shorter

Reply:
I'm not sure if that approach would work. If vite uses multiple processes to compile files, you'd have collisions